clang/test/dpct/helper_files_ref/include/device.hpp

```diff
@@ -276,6 +286,36 @@ class device_ext : public sycl::device {
         get_info<sycl::info::device::global_mem_size>());
     prop.set_local_mem_size(get_info<sycl::info::device::local_mem_size>());
 
+#if (defined(SYCL_EXT_INTEL_DEVICE_INFO) && SYCL_EXT_INTEL_DEVICE_INFO >= 6)
+    if (this->has(sycl::aspect::ext_intel_memory_clock_rate)) {
+      prop.set_memory_clock_rate(
+          this->get_info<sycl::ext::intel::info::device::memory_clock_rate>());
+    } else {
+      std::cerr << "Querying ext_intel_device_info_memory_clock_rate is not "
+                   "supported "
+                   "by the device"
+                << std::endl;
+      std::cerr << "memory_clock_rate default value is 0." << std::endl;
+    }
+    if (this->has(sycl::aspect::ext_intel_memory_bus_width)) {
+      prop.set_memory_bus_width(
+          this->get_info<sycl::ext::intel::info::device::memory_bus_width>());
+    } else {
+      std::cerr
+          << "Querying ext_intel_device_info_memory_bus_width is not supported "
+             "by the device"
+          << std::endl;
+      std::cerr << "memory_bus_width default value is 64." << std::endl;
+    }
+#else
+    std::cerr
```

Comment:
Can this be a compile-time warning?

Reply:
no, the hardware information is obtained at runtime